jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/LocalBaseImageSteps.java

```diff
@@ -124,26 +124,47 @@ static boolean isGzipped(Path path) throws IOException {
         }
 
         return cacheDockerImageTar(
-            buildContext, tarPath, progressEventDispatcher.newChildProducer());
+            buildContext,
+            tarPath,
+            progressEventDispatcher.newChildProducer(),
+            tempDirectoryProvider);
       }
     };
   }
 
-  static Callable<LocalImage> retrieveTarImageStep(
+  static Callable<LocalImage> retrieveTarLayersStep(
       BuildContext buildContext,
       ProgressEventDispatcher.Factory progressEventDispatcherFactory,
-      Path tarPath) {
-    return () -> cacheDockerImageTar(buildContext, tarPath, progressEventDispatcherFactory);
+      Path tarPath,
+      TempDirectoryProvider tempDirectoryProvider) {
+    return () ->
+        cacheDockerImageTar(
+            buildContext, tarPath, progressEventDispatcherFactory, tempDirectoryProvider);
+  }
+
+  static Callable<ImageAndAuthorization> retrieveImageAndAuthorizationStep(
```

Comment:
I think the method name is misleading, since it doesn't try retrieving something but just building an `Image` instance from existing scalar values. `returnImageAndAuthorizationStep`?